numpy/f2py/rules.py

```diff
@@ -1081,8 +1083,10 @@
 \t\t\twhile ((_i = nextforcomb()))
 \t\t\t\t#varname#[capi_i++] = #init#; /* fortran way */
 \t\t} else {
-\t\t\tif (!PyErr_Occurred())
-\t\t\t\tPyErr_SetString(#modulename#_error,\"Initialization of #nth# #varname# failed (initforcomb).\");
+\t\t\tPyObject *exc, *val, *tb;
+\t\t\tPyErr_Fetch(&exc, &val, &tb);
+\t\t\tPyErr_SetString(#modulename#_error,\"Initialization of #nth# #varname# failed (initforcomb).\");
+\t\t\tnpy_PyErr_ChainExceptionsCause(exc, val, tb);
```

Comment:
Possible I have a missing include here, wasn't quite able to work out how to add `npy_3kcompat.h` to the include list. Let's see if its already there...

Reply:
`ImportError: /tmp/tmp4n87gp4z/_test_ext_module_5427.cpython-36m-i386-linux-gnu.so: undefined symbol: npy_PyErr_ChainExceptionsCause`

Reply:
I am a bit confused on this. Aren't these new models generated by f2py. And in that case they cannot rely on (private?) numpy API that is not available in older numpy versions?

Reply:
Hmm, but then... it works, is fortranobject.so shipped? Have to dig a bit I guess. And as for version, we have backward compatibility, not forward compatibility, so its OK if someone wants to ship compiled modules that they have to compile with the oldest f2py/numpy.

Reply:
My very lacking understanding was that f2py was a build tool, and in practice all that's ever shipped is a script that invokes it, or a binary built with it - in both cases, all the needed functions are inlined or public API.

Reply:
What you're perhaps missing is that fortranobjecr.h is included by the generated code, which is why I added the include there.